src/app/api/reminder/route.ts

```diff
@@ -0,0 +1,241 @@
+    const {
+      id,
+      setById,
+      setForId,
+      scheduledAt,
+      startDate,
+      endDate,
+      description,
+      type,
+      intervalInDays,
+      isReminderActive
+    } = await request.json();
+
+    if (!id) {
+      return new Response(
+        JSON.stringify({ error: "Reminder ID is required" }),
+        {
+          status: HttpStatusCode.BadRequest,
+          headers: { "Content-Type": "application/json" }
+        }
+      );
+    }
+
+    if (!["ALIGNER", "APPOINTMENT", "OTHER"].includes(type)) {
+      return new Response(JSON.stringify({ error: "Invalid type provided" }), {
+        status: HttpStatusCode.BadRequest,
+        headers: { "Content-Type": "application/json" }
+      });
+    }
+
+    const updatedReminder = await prisma.reminder.update({
+      where: { id },
+      data: {
+        setById,
+        setForId,
+        scheduledAt: new Date(scheduledAt),
+        startDate: new Date(startDate),
+        endDate: new Date(endDate),
+        description,
+        type,
+        intervalInDays,
+        isReminderActive
+      }
+    });
+
+    // Send email when a reminder is updated
+    await sendEmail("Reminder Updated", "updated", {
+      firstName: "Patient",
+      description,
+      startDate: startDate,
+      endDate: endDate,
+      type,
+      intervalInDays
+    });
+
+    return new Response(JSON.stringify(updatedReminder), {
+      status: HttpStatusCode.Ok,
+      headers: { "Content-Type": "application/json" }
+    });
+  } catch (error) {
+    return new Response(JSON.stringify({ error: getErrorMessage(error) }), {
+      status: HttpStatusCode.InternalServerError,
+      headers: { "Content-Type": "application/json" }
+    });
+  }
+}
+
+// DELETE: Delete an existing reminder
+export async function DELETE(request: Request) {
+  try {
+    const { id, description, startDate, endDate, type, intervalInDays } =
+      await request.json();
+
+    if (!id) {
+      return new Response(
+        JSON.stringify({ error: "Reminder ID is required" }),
+        {
+          status: HttpStatusCode.BadRequest,
+          headers: { "Content-Type": "application/json" }
+        }
+      );
+    }
+
+    await prisma.reminder.delete({
+      where: { id }
+    });
+
+    // Send email when a reminder is deleted
+    await sendEmail("Reminder Deleted", "deleted", {
+      firstName: "Patient",
+      description: `Your Reminder has been deleted successfully.`,
+      startDate: startDate,
+      endDate: endDate,
+      type,
+      intervalInDays
+    });
+
+    return new Response(
+      JSON.stringify({ message: "Reminder deleted successfully" }),
+      {
+        status: HttpStatusCode.Ok,
+        headers: { "Content-Type": "application/json" }
+      }
+    );
+  } catch (error) {
+    return new Response(JSON.stringify({ error: getErrorMessage(error) }), {
+      status: HttpStatusCode.InternalServerError,
+      headers: { "Content-Type": "application/json" }
+    });
+  }
+}
```

Comment:
`PATCH` and `DELETE` operations should be performed on a specific reminder only, so these two endpoints should in file `api/reminder/[id]/route.ts`. The param `id` refers to the reminder ID. Refer to the `/users` endpoint.